Tools/auterion/upload_skynode.sh

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env bash
+
+DIR="$(dirname $(readlink -f $0))"
+DEFAULT_AUTOPILOT_HOST=10.41.1.1
+DEFAULT_AUTOPILOT_PORT=33333
+DEFAULT_AUTOPILOT_USER=auterion
+
+for i in "$@"
+do
+    case $i in
+        --file=*)
+        PX4_BINARY_FILE_ARGUMENT=" -f ${i#*=}"
+        ;;
+        --default-ip=*)
+        DEFAULT_AUTOPILOT_HOST="${i#*=}"
+        ;;
+        --default-port=*)
+        DEFAULT_AUTOPILOT_PORT="${i#*=}"
+        ;;
+        --default-user=*)
+        DEFAULT_AUTOPILOT_USER="${i#*=}"
+        ;;
+        --revert)
+        REVERT_AUTOPILOT_ARGUMENT=" -r"
+        ;;
+        --wifi)
+        DEFAULT_AUTOPILOT_HOST=10.41.0.1
+        ;;
+        *)
+            # unknown option
+        ;;
+    esac
+done
+
+# allow these to be overridden
+[ -z "$AUTOPILOT_HOST" ] && AUTOPILOT_HOST=$DEFAULT_AUTOPILOT_HOST
+[ -z "$AUTOPILOT_PORT" ] && AUTOPILOT_PORT=$DEFAULT_AUTOPILOT_PORT
+[ -z "$AUTOPILOT_USER" ] && AUTOPILOT_USER=$DEFAULT_AUTOPILOT_USER
+
+ARGUMENTS=()
+ARGUMENTS+=(-d $AUTOPILOT_HOST)
+ARGUMENTS+=(-p $AUTOPILOT_PORT)
+ARGUMENTS+=(-u $AUTOPILOT_USER)
+ARGUMENTS+=($PX4_BINARY_FILE_ARGUMENT)
```

Comment:
This is still incorrect with spaces. You need to use `ARGUMENTS+=(-f "$PX4_BINARY_FILE")`

Reply:
Well the file argument is optional because it's not needed when reverting. I can of course fill the variable with the file name only and then conditionally also add the -f before if the name is not empty. I thought it's cleaner that way but will check this option 🙏

Reply:
I found https://stackoverflow.com/a/20306982/6326048 and learned by using `:+` I can make an entire part optional based on if a variable is set. It also works on zsh. I added this to b9c4dc8b97c81af9fdd2bea41554a33f3e3cd561.